bin/shutter

```diff
@@ -6274,6 +6274,15 @@ sub STARTUP {
 		#start postprocessing here
 
 		#...successfully???
+		
+		# $screenshot is undefined if the selection width or height is zero
+		# Actually this should be handled by Shutter::Screenshot::Error but fails for an area with zero width or height for some reason
```

Comment:
This is because `$screenshot` is either an actual screenshot, or a single digit. This design choice is bad, but that's what we have here.

Currently Shutter::Screenshot::Error treats "undefined" as not an error.

In this case, is `$screenshot` here `undef` or `0`?

Reply:
It's `undef` indeed.  So, maybe we can make `Shutter::Screenshot::Error` check if `$screenshot` is undefined and make a suiting error handling in this case?

Reply:
Or alternatively, return an error code instead of undef from the screenshot tool in this case. Not sure which of them is better, if to ignore an obviously better solution of not mixing the error in the screenshot type itself.

Reply:
I had a look into it. The original problem is the following: If we have a selection with non-zero height and width and confirm it by double-clicking outside of the selection, the width (and possibly also height) for the selection is being considered as zero, thus giving 

```
DEBUG: s=$VAR1 = {
          'y' => 761,
          'width' => 0,
          'height' => 0,
          'x' => 1037
        };
```

As far as I can see, the variable `$s` in the `take_screenshot` subroutine from `Shutter::Screenshot::SelectorAdvanced`
is returned by `Gtk3::ImageView::Tool:Selector->get_selection`, so it might be not actually a Shutter bug. Instead of error handling, it would be better, if gtk3-imageview determined the width and height of the selection, even if confirming with the cursor outside of the selection.

Reply:
I added some debug output into `Gtk3::ImageView->SET_PROPERTY`:

```
DEBUG: oldval=$VAR1 = {
          'width' => 460,
          'y' => '282.303771972656',
          'x' => '396.209259033203',
          'height' => 317
        };

DEBUG: newval =$VAR1 = {
          'height' => '317.049499511719',
          'x' => '396.209259033203',
          'y' => '282.303771972656',
          'width' => '462.430145263672'
        };

DEBUG: oldval=$VAR1 = {
          'height' => 317,
          'x' => '396.209259033203',
          'y' => '282.303771972656',
          'width' => 462
        };

DEBUG: newval =$VAR1 = {
          'y' => '282.303771972656',
          'width' => '462.430145263672',
          'height' => '317.049499511719',
          'x' => '396.209259033203'
        };

DEBUG: oldval=$VAR1 = {
          'y' => '282.303771972656',
          'width' => 462,
          'height' => 317,
          'x' => '396.209259033203'
        };
# here the cursor moves outside of the selection (x=1084, y=702) and the height and width become zero:
DEBUG: newval =$VAR1 = { 
          'height' => '0',
          'x' => '1084.86181640625',
          'y' => '701.156372070312',
          'width' => 0
        };

```
But I cannot find where this subroutine is called and where it takes its arguments from (in particular `$newval` which has zero width and height at some point).

Reply:
So, I guess, I figured it out: If the cursor is outside of the selection when double clicking, the cursor type is set to "crosshair" at https://metacpan.org/release/ASOKOLOV/Gtk3-ImageView-10/source/lib/Gtk3/ImageView/Tool/Selector.pm#L72 This leads to a width and height of zero, because the two coordinates are set to equal value for this cursor type later: https://metacpan.org/release/ASOKOLOV/Gtk3-ImageView-10/source/lib/Gtk3/ImageView/Tool/Selector.pm#L144 I wonder why it is made this way.